components/script/dom/htmlinputelement.rs

```diff
@@ -1231,9 +1233,20 @@ impl Activatable for HTMLInputElement {
     }
 }
 
-fn filter_from_accept(_s: DOMString) -> Vec<FilterPattern> {
-    /// TODO: it means not pattern restriction now
-    /// Blocked by https://github.com/cybergeek94/mime_guess/issues/19
-    vec![]
-}
+// https://html.spec.whatwg.org/multipage/#attr-input-accept
+fn filter_from_accept(s: &DOMString) -> Vec<FilterPattern> {
+    let mut filter = vec![];
+    for p in split_commas(s) {
+        if let Some('.') = p.chars().nth(0) {
+            filter.push(FilterPattern(p[1..].to_string()));
+        } else {
+            if let Some(exts) = mime_guess::get_mime_extensions_str(p) {
```

Comment:
There might be a better way of doing this by directly telling the OS the mime type. However, since it's not in the tfd api yet, we can just merge this for now.